### PR TITLE
Use CMake's FetchContent module to (optionally) download and build FMILibrary

### PR DESCRIPTION
**Pull request by prashanthr05**
Uses CMake 3.11 FetchContent.cmake option to download and compile FMILibrary , if find_package fails to find FMILibrary.

The minimum cmake version is still 3.5. However, the cmake 3.11 specific files are copied into the repository's cmake/cmake-3.11 folder.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -49,13 +53,17 @@ if (gazebo_VERSION_MAJOR LESS 7.0)
 endif()
 
 find_package(FMILibrary REQUIRED)
+if (NOT FMILibary_FOUND})
```
Comment (traversaro) [inline CMakeLists.txt]: Typo, it should be `NOT FMILibrary_FOUND`

**Comment by traversaro** [comment]
The compromise we found in iDynTree/YARP/ICUB is: there is always an option `USE_<dep>` or `USE_SYSTEM_<dep>` , and its **default** value is initialized to `ON` or `OFF` depending on wherever the dep was found in the system or not. However, if the user/superbuild specifies the option, the option is observed.  See https://github.com/robotology/idyntree/issues/322 and https://github.com/robotology/idyntree/pull/323 .

**Comment by traversaro** [comment]
@prashanthr05 I edited the title to be more descriptive.

File: README.md [hunk 2]
```diff
@@ -28,10 +28,12 @@ gazebo-fmi depends on
...
 
-We recommend to install Gazebo as described  in [official documentation](http://gazebosim.org/tutorials?cat=install),
-while for FMILibrary the easiest option is to compile it as any CMake project and then add its installation prefix to [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/v3.10/variable/CMAKE_PREFIX_PATH.html).
+We recommend to install Gazebo as described  in [official documentation](http://gazebosim.org/tutorials?cat=install).
+For FMILibrary, one option is to compile it as any CMake project and then add its installation prefix to [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/v3.10/variable/CMAKE_PREFIX_PATH.html).
 See [CGold guide](https://cgold.readthedocs.io/en/latest/first-step.html) if you need some details on how to build a CMake project.
 
+An easier option is provided by the gazebo-fmi repository which downloads and compiles the FMILibrary internally within its build folder using CMake's FetchContent option. This is performed when FMILibrary package is not already existing in the system or if the FMILibrary related environment variable (`FMI_ROOT`) is not set, making gazebo-fmi unable to find FMILibrary package.
+
```
Comment (traversaro) [inline README.md]: `FMI_ROOT` --> `FMILibrary_ROOT`

**Comment by prashanthr05** [comment]
The requested changes addressed in this [commit](https://github.com/robotology/gazebo-fmi/pull/23/commits/6393d1c20b14a337e3cbc5dbabbef7f517e80835)

**Comment by prashanthr05** [comment]
Feature #2

File: CMakeLists.txt [hunk 3]
```diff
@@ -48,14 +52,20 @@ if (gazebo_VERSION_MAJOR LESS 7.0)
     message(FATAL_ERROR "Your Gazebo version is older than Gazebo 7.0. Gazebo Yarp plugins are supported with gazebo versions >= 7.0. Please update to a newer version")
 endif()
 
-find_package(FMILibrary REQUIRED)
+option(USES_SYSTEM_FMILIBRARY "Choose to use system FMILibrary OR download and compile using FetchContent" ON)
+if (USES_SYSTEM_FMILIBRARY)
+    find_package(FMILibrary REQUIRED)
+else()
+   message("Could not find FMILIbrary package. Downloading it using FetchContent...")
```
Comment (traversaro) [inline CMakeLists.txt]: I would remove this message.

**Comment by traversaro** [comment]
Ok, I just noticed that the default value is **not** to use the Fetched library. In this case, feel free to decide to stick with the current behavior or to adopt the logic that I suggested in the previous comment.

**Comment by prashanthr05** [comment]
Addressed the changes. Thanks @traversaro

**Comment by traversaro** [comment]
Thanks!

**Pull request merged**